### PR TITLE
Use Swift's time types

### PR DESCRIPTION
**Pull request by yim-lee**
Resolves https://github.com/apple/swift-distributed-actors/issues/955

### CODE REVIEW

File: Sources/DistributedActors/Time.swift [hunk 1]
```diff
@@ -18,116 +18,72 @@ import struct Foundation.Date
...
     /// - parameters:
-    ///     - amount: the amount of hours this `TimeAmount` represents.
-    /// - returns: the `TimeAmount` for the given amount.
-    public static func hours(_ amount: Value) -> TimeAmount {
-        TimeAmount(amount * 1000 * 1000 * 1000 * 60 * 60)
+    ///     - amount: the amount of hours this `Duration` represents.
+    /// - returns: the `Duration` for the given amount.
+    public static func hours(_ amount: Value) -> Duration {
+        .nanoseconds(amount * 1000 * 1000 * 1000 * 60 * 60)
     }
```
Comment (ktoso) [inline Sources/DistributedActors/Time.swift:75]: This always felt a bit silly to be dropped off from the official type... thanks for adding them

I wonder if we "have to" make them internal...

Comment (ktoso) [reply Sources/DistributedActors/Time.swift:75]: (all the .hours / .seconds thingies i mean)

File: Sources/DistributedActors/Time.swift [hunk 2]
```diff
@@ -209,187 +165,45 @@ extension TimeAmount: CustomStringConvertible, CustomPrettyStringConvertible {
...
-/// - note: `Deadline` should not be used to represent a time interval
-public struct Deadline: Equatable, Hashable {
-    // TODO(swift): To be replaced by types that upcoming Swift versions might gain, as discussed in
-    //              [Pitch] Clock, Instant, Date, and Duration https://forums.swift.org/t/pitch-clock-instant-date-and-duration/52451
-
-    public typealias Value = Int64
-
-    /// The nanoseconds since boot representation of the `Deadline`.
-    public let uptimeNanoseconds: Value
+public typealias Deadline = ContinuousClock.Instant
```
Comment (ktoso) [inline Sources/DistributedActors/Time.swift:191]: Probably another case where people might be upset at us declaring a public typealias... might have to remove?

File: Sources/DistributedActors/Time.swift [hunk 3]
```diff
@@ -209,187 +165,45 @@ extension TimeAmount: CustomStringConvertible, CustomPrettyStringConvertible {
...
+        .now() - Duration.effectivelyInfinite
     }
 
-    public static let distantPast = Deadline(0)
-    public static let distantFuture = Deadline(Value.max)
-
-    public static func now() -> Deadline {
-        uptimeNanoseconds(Deadline.Value(DispatchTime.now().uptimeNanoseconds))
+    public static var distantFuture: ContinuousClock.Instant {
+        .now() + Duration.effectivelyInfinite
```
Comment (ktoso) [inline Sources/DistributedActors/Time.swift:199]: are these correct?

File: Sources/DistributedActors/Time.swift [hunk 4]
```diff
@@ -209,187 +165,45 @@ extension TimeAmount: CustomStringConvertible, CustomPrettyStringConvertible {
...
     }
 
-    public static func uptimeNanoseconds(_ nanoseconds: Value) -> Deadline {
-        Deadline(nanoseconds)
+    public static func now() -> ContinuousClock.Instant {
+        .now
     }
```
Comment (ktoso) [inline Sources/DistributedActors/Time.swift:204]: can probably remove `.now()` in favor of using `.now`

File: Sources/DistributedActors/Time.swift [hunk 5]
```diff
@@ -398,7 +212,7 @@ extension Deadline {
         self.hasTimeLeft(until: .now())
     }
 
-    public func hasTimeLeft(until: Deadline) -> Bool {
+    public func hasTimeLeft(until: ContinuousClock.Instant) -> Bool {
         !self.isBefore(until)
     }
```
Comment (ktoso) [inline Sources/DistributedActors/Time.swift:217]: I like those... let's keep -- I wonder if people might be upset if we make them public though 🤔

**Review by ktoso — APPROVED** [review]
Looks great, only concern about potentially needing to mark extensions on Duration and the typealias `internal` so folks don't get upset we're adding public API to "not our type".

File: Sources/DistributedActors/Time.swift (hunk 1 above)
Comment (yim-lee) [reply Sources/DistributedActors/Time.swift:75]: Made these internal

File: Sources/DistributedActors/Time.swift (hunk 3 above)
Comment (yim-lee) [reply Sources/DistributedActors/Time.swift:199]: Hmm I think so (and there are tests)? It used to be `Deadline(Int64.max)`.

File: Sources/DistributedActors/Time.swift (hunk 5 above)
Comment (yim-lee) [reply Sources/DistributedActors/Time.swift:217]: Made these internal

File: Sources/DistributedActors/Time.swift (hunk 2 above)
Comment (yim-lee) [reply Sources/DistributedActors/Time.swift]: removed

File: Sources/DistributedActors/Time.swift (hunk 3 above)
Comment (ktoso) [reply Sources/DistributedActors/Time.swift]: Strike all that I tried it out and it works out since we cap inside the `.nanoseconds`

Thank you!

**Pull request merged**